views/js/test/qtiCommonRenderer/interactions/textEntry/test.js

```diff
@@ -0,0 +1,192 @@
+define([
+    'jquery',
+    'lodash',
+    'taoQtiItem/runner/qtiItemRunner',
+    'json!taoQtiItem/test/samples/json/text-entry-noconstraint.json',
+    'json!taoQtiItem/test/samples/json/text-entry-length.json',
+    'json!taoQtiItem/test/samples/json/text-entry-pattern.json'
+], function ($, _, qtiItemRunner, textEntryData, textEntryLengthConstrainedData, textEntryPatternConstrainedData) {
+    'use strict';
+
+    var runner;
+    var fixtureContainerId = 'item-container';
+    var outsideContainerId = 'outside-container';
+
+    module('Text Entry Interaction', {
+        teardown: function () {
+            if (runner) {
+                runner.clear();
+            }
+        }
+    });
+
+    function getTooltipContent($input){
+        var qtip = getTooltip($input);
+        if(qtip){
+            return qtip.find('.qtip-content').html();
+        }
+    }
+
+    function getTooltip($input){
+        var qtip = $input.data('qtip');
+        if(qtip && qtip.tooltip && qtip.tooltip.length){
+            return $(qtip.tooltip[0]);
+        }
+    }
+
+    QUnit.asyncTest('Lenght constraint', function (assert) {
+
+        var $container = $('#' + fixtureContainerId);
+
+        assert.equal($container.length, 1, 'the item container exists');
+        assert.equal($container.children().length, 0, 'the container has no children');
+
+        runner = qtiItemRunner('qti', textEntryLengthConstrainedData)
+            .on('render', function () {
+                var $input = $container.find('.qti-interaction.qti-textEntryInteraction');
+
+                assert.equal($input.length, 1, 'the container contains a text entry interaction .qti-textEntryInteraction');
+
+                $input.val('');
+                $input.focus();
+                assert.equal(getTooltipContent($input), '5 characters allowed', 'the instruction message is correct');
+                assert.ok(getTooltip($input).is(':visible'), 'info tooltip is visible');
+
+                $input.val('123');
+                $input.keyup();
+                assert.equal(getTooltipContent($input), '3/5', 'the instruction message is correct');
+                assert.ok(getTooltip($input).is(':visible'), 'info tooltip is visible');
+
+                $input.val('12345');
+                $input.keyup();
+                assert.equal(getTooltipContent($input), '5/5', 'the warning message is correct');
+                assert.ok(getTooltip($input).is(':visible'), 'warning tooltip is visible');
+                assert.ok($input.hasClass('maxed'), 'has state maxed');
+
+                $input.val('1234');
+                $input.keyup();
+                assert.equal(getTooltipContent($input), '4/5', 'the instruction message is correct');
+                assert.ok(getTooltip($input).is(':visible'), 'info tooltip is visible');
+                assert.ok(!$input.hasClass('maxed'), 'has state maxed removed');
+
```

Comment:
Interesting yes, but unfortunately, I didn't find a way to do this. Indeed if we simulate a key input event in an input, the event is fired but the input value is not updated so no way to really simulate a use typing one by one, which is a pity.